src/audio/volume/Kconfig

```diff
@@ -76,4 +76,25 @@ config COMP_GAIN
 	  This option enables gain to change volume. It works
 	  as peak volume without updating peak vol to host
 
+config VOLUME_HIFI4
+    bool
+    default y if ACE
+    default n
+    help
+        This option used to enable HIFI4 optimized volume code for sof build
+
+config VOLUME_HIFI3
+    bool
+    default y if CAVS_VERSION_2_5
+    default n
+    help
+        This option used to enable HIFI3 optimized volume code for sof build
+
+config VOLUME_GENERIC
+    bool
+    default y if !VOLUME_HIFI3 && !VOLUME_HIFI4
+    default n
+    help
+        This option used to enable volume generic code for sof build
```

Comment:
it makes no sense to enable more than one of these, right? They're mutually-exclusive? should this be a `choice`?

Reply:
check core-isa.h generated during build, XCHAL_HAVE_HIFI3 and XCHAL_HAVE_HIFI4 are enabled together.

I am ok with choice, I make this PR as review status because I want to listen whether enable HIFI version based on platform is ok, I don't have other good ideas, if config it in Kconfig, this is the only way I can work out.

Reply:
@lyakh , we can' t use choice, as this is not manually determined, it is determined by the toolchain, which is further determined by the platform config, please check this version, I add a new build envionment varaible.

Reply:
but actually, I think we can use a choice with default value depend on tool chain version.